source_http.go

```diff
@@ -86,10 +86,10 @@ func (s *HTTPImageSource) setAuthorizationHeader(req *http.Request, ireq *http.R
 	}
 }
 
-func (s *HTTPImageSource) setCustomHeaders(req *http.Request, ireq *http.Request) {
-	headers := s.Config.CustomHeaders
+func (s *HTTPImageSource) setForwardHeaders(req *http.Request, ireq *http.Request) {
+	headers := s.Config.ForwardHeaders
 	for _, header := range headers {
-		if header != "" {
+		if header != "" && ireq.Header.Get(header) != "" && req.Header.Get(header) == "" {
```

Comment:
This change breaks HTTP expectations. Headers can actually have an [empty value](https://http2.github.io/http2-spec/#Http2SettingsHeader), in fact HTTP/2 uses it as part of its upgrade process.

I'm not sure if this is actually a problem in this use-case, but I also think it's fine. Fewer code is better in this situation I think. So if you stick to the `if header != "" {` here, it should be good.